src/tiffvisitor_int.cpp

```diff
@@ -462,6 +463,78 @@ namespace Exiv2 {
         }
     } // TiffMetadataDecoder::decodeIptc
 
+    static const TagInfo* findTag(const TagInfo* pList,uint16_t tag)
+    {
+        while ( pList->tag_ != 0xffff && pList->tag_ != tag ) pList++;
+        return  pList->tag_ != 0xffff  ? pList : NULL;
+    }
+
+    void TiffDecoder::decodeCanonAFInfo(const TiffEntryBase* object) {
+        // report Exif.Canon.AFInfo as usual
+        TiffDecoder::decodeStdTiffEntry(object);
+        if ( !object->pValue()->count() || object->pValue()->typeId() != unsignedShort ) return; // no data!
+
+        // create vector of signedShorts from unsignedShorts in Exif.Canon.AFInfo
+        std::vector<int16_t>  ints;
+        std::vector<uint16_t> uint;
+        for (int i = 0; i < object->pValue()->count(); i++) {
+            ints.push_back((int16_t) object->pValue()->toLong(i));
+            uint.push_back((uint16_t) object->pValue()->toLong(i));
+        }
+        // Check this is AFInfo2 (ints[0] = bytes in object)
+        if ( ints[0] != object->pValue()->count()*2 ) return ;
+
+        std::string familyGroup(std::string("Exif.") + groupName(object->group()) + ".");
+
+        const uint16_t nPoints = uint[2];
```

Comment:
There is no guarantee that `uint` contains at least 3 elements.
```suggestion
        const uint16_t nPoints = uint.at(2);
```

Reply:
Modified the `insufficient data` test on line 475 (great catch, well done Dan).
Adopted `.at(2)`.